ts/features/messages/store/reducers/thirdPartyById.ts

```diff
@@ -63,6 +54,47 @@ export const thirdPartyByIdReducer = (
       return toError(action.payload.id, state, action.payload.error);
     case getType(reloadAllMessages.request):
       return initialState;
+    case getType(populateStoresWithEphemeralAarMessageData):
+      const {
+        iun,
+        pnServiceID,
+        subject,
+        mandateId,
+        thirdPartyMessage,
+        markDown,
+        fiscalCode
+      } = action.payload;
+
+      return toSome(iun, state, {
+        kind: "AAR",
+        mandateId,
+        created_at: new Date(),
+        third_party_message: thirdPartyMessage,
+        id: iun,
+        fiscal_code: fiscalCode as FiscalCode,
+        sender_service_id: pnServiceID,
+        content: {
+          third_party_data: {
+            has_attachments: true,
+            has_precondition: HasPreconditionEnum.ALWAYS,
+            id: iun
+          },
+          markdown: markDown,
+          subject
+        }
+      });
+
+    case getType(terminateAarFlow):
+      const newState = _.pickBy(state, (value, _key) =>
+        pipe(
```

Comment:
Also here I would suggest 
`value != null && pot.isSome(value) && !isEphemeralAARThirdPartyMessage(value.value)`
instead of a pipe.

Take also in consideration to extract the entire inner function into a standalone one, so that the code becomes
`const newState = _.pickBy(state, theExtractedFunction)`